bin/cache_repo

```diff
@@ -9,11 +9,11 @@ rm -rf "/tmp/$DATE"
 mkdir -p "/tmp/$DATE"
 
 # Download the repo
-git clone --mirror $BUILDKITE_REPO "/tmp/$DATE/$BUILDKITE_PIPELINE_SLUG.git"
+git clone --mirror "$BUILDKITE_REPO" "/tmp/$DATE/$BUILDKITE_PIPELINE_SLUG.git"
 
 # Create the tarball
 TAR_NAME=$DATE.git.tar
-tar -C "/tmp/$DATE" -cvf $TAR_NAME $BUILDKITE_PIPELINE_SLUG.git
+tar -C "/tmp/$DATE" -cvf "$TAR_NAME" "$BUILDKITE_PIPELINE_SLUG".git
 
 # Copy the file to S3
-aws s3 cp $TAR_NAME s3://$AWS_BUCKET/$BUILDKITE_ORGANIZATION_SLUG/$BUILDKITE_PIPELINE_SLUG/$TAR_NAME
+aws s3 cp "$TAR_NAME" s3://"$AWS_BUCKET"/"$BUILDKITE_ORGANIZATION_SLUG"/"$BUILDKITE_PIPELINE_SLUG"/"$TAR_NAME"
```

Comment:
This is what the `shellcheck -f diff` suggested, but I think quoting the whole URL at once, in one big quotation, would work as well? 🤷

Reply:
Yeah, IINM this would actually break the URL since it would interpret the `"` as components of the URL, no?

Reply:
Nope the way this diff is done won't break, it's totally legal in shell to have quotes in the middle of a string and it will still eat those quotes, not keep them verbatim. So this will have the intended behavior and the same behavior as is we quoted the whole strings as once.

In other words, in bash all these are equivalent and won't eg any literal quote:

```
echo "Hello foo/bar/baz"
echo "Hello "foo/"bar"/"baz"
echo Hello\ foo/bar/baz
echo He"llo fo"o/b"ar/baz"
```

The only case when a double quote would be kept as a literal even in the middle of a string is if that double quote is either escaped with a backslash, or if it is itself inside a single-quoted string:

```
echo '"Hello", she said.'
echo \"Hello\",\ she\ said.
```

---

So TL;DR:
 - the change made by shellcheck here is totally valid and won't introduce literal quotes
 - quoting the whole URL at once will produce the exact same result. Which since it's arguably more readable for humans, is likely what we should do instead. But if we don't and keep that suggested diff instead that will still work.

Reply:
Addressed in af3e002